### PR TITLE
Use docker toolchain rule in incremental_load

### PR DESCRIPTION
**Pull request by smukherj1**
Add the toolchain_type as a toolchain field to import, bundle
and image that use incremental_load under the hood

### CODE REVIEW

File: tools/BUILD [hunk 1]
```diff
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-package(default_visibility = ["//visibility:private"])
+package(default_visibility = ["//visibility:public"])
```
Comment (nlopezgi) [inline tools/BUILD]: can we just increase visibility of the toolchain_type_docker target only

Comment (smukherj1) [reply tools/BUILD]: Done

File: toolchains/docker/BUILD [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+toolchain_type(
+    name = "toolchain_type",
+)
+
+# Default docker toolchain that expects the 'docker' executable
+# to be in the PATH
+docker_toolchain(
+    name = "default_toolchain_impl",
+    tool_path = "docker",
+    visibility = ["//visibility:public"],
```
Comment (nlopezgi) [inline toolchains/docker/BUILD]: can we restrict the visibility of this target?

Comment (smukherj1) [reply toolchains/docker/BUILD]: Nope. This is actually the rule that is being accessed by rules that use this toolchain_type. Making this private will give visibility errors

Comment (nlopezgi) [reply toolchains/docker/BUILD]: got it, sgtm

**Pull request merged**